### PR TITLE
OCPBUGS-44372: PPC: skip comparing ProcessorCore.Index between NUMA cores 

### PR DESCRIPTION
**Pull request by shajmakh**
ProcessorCore.Index indicates the zero-based index of the core in the
Cores slice. While core might be shown in a different order, they can still
be equivalent. See: jaypipes/ghw#346.

Adjust the equality check to skip this field to fix this:

```
  Error: targeted nodes differ: nodes host1.development.lab and host2.development.lab have different topology: the CPU cores differ: processor core #20 (2 threads), logical processors [2 66] vs processor core #20 (2 threads), logical processors [2 66]
```

And add a unit test to cover this scenario and the sortTopology function. 

Signed-off-by: Shereen Haj <shajmakh@redhat.com>

### CODE REVIEW

**Comment by shajmakh** [comment]
/cherry-pick release-4.17 release-4.16 release-4.15 release-4.14 release-4.13 release-4.12

File: pkg/performanceprofile/profilecreator/profilecreatorfuncs_test.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/jaypipes/ghw/pkg/cpu"
+	"github.com/jaypipes/ghw/pkg/topology"
+)
+
+func TestSortTopology(t *testing.T) {
```
Comment (ffromani) [inline pkg/performanceprofile/profilecreator/profilecreatorfuncs_test.go:11]: why do we need to extract the `sortTopology` helper (and test a private function :\ )? Can we test the public `SortedTopology()` public function?

**Review by ffromani — COMMENTED** [review]
/approve

LGTM once inline comments are addressed

File: pkg/performanceprofile/profilecreator/profilecreatorfuncs_test.go (hunk 1 above)
Comment (shajmakh) [reply pkg/performanceprofile/profilecreator/profilecreatorfuncs_test.go]: followup PR: https://github.com/openshift/cluster-node-tuning-operator/pull/1217

File: pkg/performanceprofile/profilecreator/profilecreator_test.go [hunk 2]
```diff
@@ -1474,6 +1474,12 @@ var _ = Describe("PerformanceProfileCreator: Test Helper Function ensureSameTopo
 			err := ensureSameTopology(&topology1, &topology2)
 			Expect(err).To(HaveOccurred())
 		})
+		It("same cores with different indices should still considered equivalent", func() {
+			topology2.Nodes[0].Cores[0].Index = 1
+			topology2.Nodes[0].Cores[1].Index = 0
```
[the remark this answers is not in the document]
Comment (ffromani) [reply pkg/performanceprofile/profilecreator/profilecreator_test.go]: ok, great. If this is guartanteed we don't need deepcopy.

File: pkg/performanceprofile/profilecreator/profilecreator.go [hunk 3]
```diff
@@ -730,8 +732,16 @@ func ensureSameTopology(topology1, topology2 *topology.Info) error {
 		}
 
 		for j, core1 := range cores1 {
-			if !reflect.DeepEqual(core1, cores2[j]) {
-				return fmt.Errorf("the CPU corres differ: %v vs %v", core1, cores2[j])
+			// skip comparing index because it's fine if they deffer; see https://github.com/jaypipes/ghw/issues/345#issuecomment-1620274077
```
Comment (ffromani) [inline pkg/performanceprofile/profilecreator/profilecreator.go:735]: typo: differ

**Review by ffromani — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**